src/components/SegmentedRadioGroup/SegmentedRadioGroupContext.ts

```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+
+import type {UseRadioGroupContextProps} from '../../hooks/private/useRadioGroup/types';
+
+export const SegmentedRadioGroupContext = React.createContext<UseRadioGroupContextProps>({
```

Comment:
It would be better to make this context a part of `useRadioGroup` hook, and apply it in both `SegmentedRadioGroup` and `RadioGroup`. And maybe it should be named just `RadioGroupContext`